nixos/modules/services/networking/wstunnel.nix

```diff
@@ -0,0 +1,341 @@
+        example = "wstunnel";
+      };
+
+      hostHeader = mkOption {
+        description = mdDoc "Use this as the HTTP host header instead of the real hostname. Useful for circumventing hostname-based firewalls.";
+        type = types.nullOr types.str;
+        default = null;
+      };
+
+      tlsSNI = mkOption {
+        description = mdDoc "Use this as the SNI while connecting via TLS. Useful for circumventing hostname-based firewalls.";
+        type = types.nullOr types.str;
+        default = null;
+      };
+
+      tlsVerifyCertificate = mkOption {
+        description = mdDoc "Whether to verify the TLS certificate of the server. It might be useful to set this to `false` when working with the `tlsSNI` option.";
+        type = types.bool;
+        default = true;
+      };
+
+      # The original argument name `websocketPingFrequency` is a misnomer, as the frequency is the inverse of the interval.
+      websocketPingInterval = mkOption {
+        description = mdDoc "Do a heartbeat ping every N seconds to keep up the websocket connection.";
+        type = types.nullOr types.int;
+        default = null;
+      };
+
+      upgradeCredentials = mkOption {
+        description = mdDoc "Use these credentials to authenticate during the HTTP upgrade request (Basic authorization type, `USER:[PASS]`). WARNING: Passwords specified here will be world-readable in the Nix store! To safely pass a password to the service, point the `environmentFile` option to a file containing `HTTP_PASSWORD=<your-password-here>` and set this option to `<user>:$HTTP_PASSWORD`.";
+        type = types.nullOr types.str;
+        default = null;
+      };
+
+      customHeaders = mkOption {
+        description = mdDoc "Custom HTTP headers to send during the upgrade request.";
+        type = types.attrsOf types.str;
+        default = {};
+        example = {
+          "X-Some-Header" = "some-value";
+        };
+      };
+
+      verboseLogging = mkOption {
+        description = mdDoc "Enable verbose logging.";
+        type = types.bool;
+        default = false;
+      };
+
+      runAsRoot = mkOption {
+        description = mdDoc "Run `wstunnel` as root. Useful when using `soMark`.";
+        type = types.bool;
+        default = false;
+      };
+
+      package = mkOption {
+        description = mdDoc "The `wstunnel` package to use for this instance.";
+        type = types.package;
+        default = cfg.package;
+        defaultText = literalExpression "services.wstunnel.package";
+      };
+
+      environmentFile = mkOption {
+        description = mdDoc "Environment file to be passed to the systemd service. Useful for passing secrets to the service.";
+        type = types.nullOr types.path;
+        default = null;
+        example = "/var/lib/secrets/wstunnelSecrets";
+      };
+
+      extraArgs = mkOption {
+        description = mdDoc "Extra command line arguments to pass to `wstunnel`.";
+        type = types.str;
+        default = null;
```

Comment:
This should be more an attr of str

Reply:
Done, and took the possibility of unary arguments into account